sematic/resolvers/silent_resolver.py

```diff
@@ -19,7 +20,14 @@ def _run_inline(self, future: AbstractFuture) -> None:
         self._set_future_state(future, FutureState.SCHEDULED)
         try:
             self._start_inline_execution(future.id)
-            value = future.calculator.calculate(**future.resolved_kwargs)
+            with set_context(
+                SematicContext(
+                    id=future.id,
+                    root_id=self._root_future.id,
+                    resolver_class_path=self.classpath(),
+                )
+            ):
```

Comment:
How about providing syntactic sugar to remove some of the boilerplate?
```suggestion
            with set_context(
                id=future.id,
                root_id=self._root_future.id,
                resolver=self,
            ):
```

Reply:
this `with` block only occurs in two places, so removing boilerplate isn't particularly crucial. I'd prefer to keep the code a bit simpler